src/compiler/diagnosticMessages.json

```diff
@@ -1974,7 +1974,7 @@
         "category": "Error",
         "code": 6053
     },
-    "File '{0}' must have extension '.ts' or '.d.ts'.": {
+    "File '{0}' has unsupported extension. The only supported extensions are {1}.": {

```

Comment:
did that,, then felt too hard to read; this is still grammatically correct and gives the same meaning.